jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/UsageResource.java

```diff
@@ -133,7 +133,7 @@ public Response recordUsage(final SubscriptionUsageRecordJson json,
         final Entitlement entitlement = entitlementApi.getEntitlementForId(json.getSubscriptionId(), callContext);
         if (entitlement.getEffectiveEndDate() != null) {
             final LocalDate highestRecordDate = getHighestRecordDate(json.getUnitUsageRecords());
-            if (entitlement.getEffectiveEndDate().compareTo(highestRecordDate) < 0) {
+            if (entitlement.getEffectiveEndDate().toLocalDate().compareTo(highestRecordDate) < 0) { //TODO_1375 - Converting to LocalDate for comparison, should we instead convert highestRecordDate to DateTime and use it? Also, not sure how to access internalCallContext here so I've used toLocalDate
                 return Response.status(Status.BAD_REQUEST).build();
```

Comment:
Same comment as above, should we use `DateTime` or `LocalDate` ?

Reply:
Same answer